src/box/allocator.h

```diff
@@ -0,0 +1,153 @@
+static inline void								\
+type##_##allocator_enter_delayed_free_mode(struct memtx_engine *memtx)  	\
+{										\
+	return type##_##alloc_setopt(memtx->alloc,				\
+		TYPE##_##DELAYED_FREE_MODE, true);				\
+}
+DECLARE_MEMTX_ALLOCATOR_ENTER_DELAYED_FREE_MODE(small, SMALL)
+DECLARE_MEMTX_ALLOCATOR_ENTER_DELAYED_FREE_MODE(system, SYSTEM)
+/**
+ * Global abstract method to enter delayed free mode
+ */
+typedef void (*global_enter_delayed_free_mode)(struct memtx_engine *memtx);
+
+#define DECLARE_MEMTX_ALLOCATOR_LEAVE_DELAYED_FREE_MODE(type, TYPE)		\
+static inline void								\
+type##_##allocator_leave_delayed_free_mode(struct memtx_engine *memtx)  	\
+{										\
+	return type##_##alloc_setopt(memtx->alloc,				\
+		TYPE##_##DELAYED_FREE_MODE, false);				\
+}
+DECLARE_MEMTX_ALLOCATOR_LEAVE_DELAYED_FREE_MODE(small, SMALL)
+DECLARE_MEMTX_ALLOCATOR_LEAVE_DELAYED_FREE_MODE(system, SYSTEM)
+/**
+ * Global abstract method to leave delayed free mode
+ */
+typedef void (*global_leave_delayed_free_mode)(struct memtx_engine *memtx);
+
+#define DECLARE_MEMTX_ALLOCATOR_INIT(type, SLAB_CACHE_CREATE, param)		\
+static inline void 								\
+type##_##allocator_init(struct memtx_engine *memtx, uint32_t objsize_min,	\
+			float alloc_factor)					\
+{										\
+	SLAB_CACHE_CREATE(&memtx->slab_cache, &memtx->arena);			\
+	type##_##alloc_create(memtx->alloc, &memtx->param, objsize_min,		\
+		alloc_factor);							\
+}
+DECLARE_MEMTX_ALLOCATOR_INIT(small, slab_cache_create, slab_cache)
+DECLARE_MEMTX_ALLOCATOR_INIT(system, noop_two_arg, quota)
+/**
+ * Global abstract method for allocator initialization
+ */
+typedef void (*global_allocator_init)(struct memtx_engine *memtx,
+			uint32_t objsize_min, float alloc_factor);
+
+#define DECLARE_MEMTX_ALLOCATOR_DESTROY(type, SLAB_CACHE_DESTROY)		\
+static inline void 								\
+type##_##allocator_destroy(struct memtx_engine *memtx)  			\
+{										\
+	type##_##alloc_destroy(memtx->alloc);					\
+	SLAB_CACHE_DESTROY(&memtx->slab_cache);					\
+	tuple_arena_destroy(&memtx->arena);					\
+}
+DECLARE_MEMTX_ALLOCATOR_DESTROY(small, slab_cache_destroy)
+DECLARE_MEMTX_ALLOCATOR_DESTROY(system, noop_one_arg)
+/**
+ * Global abstract method to allocator destroy
+ */
+typedef void (*global_allocator_destroy)(struct memtx_engine *memtx);
+
+#define DECLARE_MEMTX_SLAB_CACHE_CHECK(type, SLAB_CACHE_CHECK)  		\
+static inline void								\
+type##_##slab_cache_check(struct memtx_engine *memtx)				\
+{										\
+	SLAB_CACHE_CHECK(((struct type##_alloc *)(memtx->alloc))->cache);	\
+	(void)memtx;								\
+}
+DECLARE_MEMTX_SLAB_CACHE_CHECK(small, slab_cache_check)
+DECLARE_MEMTX_SLAB_CACHE_CHECK(system, noop_one_arg)
+/**
+ * Global abstract method to slab cache check
+ */
+typedef void (*global_slab_cache_check)(struct memtx_engine *memtx);
+extern global_slab_cache_check memtx_slab_cache_check;
+
+#if defined(__cplusplus)
+} /* extern "C" */
+#endif /* defined(__cplusplus) */
+
+
+#endif /* TARANTOOL_BOX_ALLOCATOR_H_INCLUDED */
```

Comment:
Git warns you about missing last empty line. Please, don't ignore.

Reply:
Fixed